### PR TITLE
Fix ec2_group for numbered protocols (GRE)

### PR DESCRIPTION
**Pull request by ryansb**
##### SUMMARY

Reported in https://github.com/ansible/ansible/issues/42740 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

ec2_group

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/integration/targets/ec2_group/tasks/numeric_protos.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - assert:
+        that:
+          - result is not changed
+    - name: Add a tag with a numeric value
+      ec2_group:
+        name: '{{ group_tmp_name }}'
+        vpc_id: '{{ vpc_result.vpc.id }}'
+        description: '{{ ec2_group_description }}'
+        tags:
+          foo: 1
```
Comment (s-hertel) [inline test/integration/targets/ec2_group/tasks/numeric_protos.yml:49]: This task and the one below need credentials

File: test/integration/targets/ec2_group/tasks/numeric_protos.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - name: Readd a tag with a numeric value
```
Comment (s-hertel) [inline test/integration/targets/ec2_group/tasks/numeric_protos.yml]: s/Readd/Read

**Comment by s-hertel** [comment]
[This task](https://github.com/ansible/ansible/blob/devel/test/integration/targets/ec2_group/tasks/main.yml#L1176) is failing repeatedly for me. 
```
TASK [ec2_group : test modifying rule and egress rule descriptions (expected changed=true)] ***********************************************
task path: /Users/shertel/Workspace/ansible/test/integration/targets/ec2_group/tasks/main.yml:1177
...
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_hIDfGf/ansible_module_ec2_group.py", line 709, in authorize
    client.authorize_security_group_ingress(GroupId=group_id, IpPermissions=ip_permissions)
  File "/Users/shertel/Workspace/ansible/venv/python2.7.13/lib/python2.7/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/shertel/Workspace/ansible/venv/python2.7.13/lib/python2.7/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (InvalidPermission.Duplicate) when calling the AuthorizeSecurityGroupIngress operation: the specified rule "peer: 1001:d00::/24, TCP, from port: 8281, to port: 8281, ALLOW" already exists

fatal: [localhost]: FAILED! => {
    "boto3_version": "1.7.52",
    "botocore_version": "1.10.78",
    "changed": false,
    "error": {
        "code": "InvalidPermission.Duplicate",
        "message": "the specified rule \"peer: 1001:d00::/24, TCP, from port: 8281, to port: 8281, ALLOW\" already exists"
    },
    "invocation": {
        "module_args": {
            ...
        }
    },
    "msg": "Unable to authorize ingress rules: [{'ToPort': 8281, 'IpProtocol': u'tcp', 'Ipv6Ranges': [{'CidrIpv6': '1001:d00::/24', 'Description': 'ipv6 rule desc 2'}], 'FromPort': 8281}]: An error occurred (InvalidPermission.Duplicate) when calling the AuthorizeSecurityGroupIngress operation: the specified rule \"peer: 1001:d00::/24, TCP, from port: 8281, to port: 8281, ALLOW\" already exists",
    "response_metadata": {
        "http_headers": {
            "connection": "close",
            "date": "Thu, 16 Aug 2018 17:38:24 GMT",
            "server": "AmazonEC2",
            "transfer-encoding": "chunked"
        },
        "http_status_code": 400,
        "request_id": "0765a398-b022-424f-a0c7-e7333d366a93",
        "retry_attempts": 0
    }
}

```

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/78997/67/tests

**Review by s-hertel — APPROVED** [review]

**Comment by s-hertel** [comment]
bot_status

**Pull request merged**